### ISSUE TITLE
cli/sql: panic in `cockroach sql --insecure` during autocomplete

### ISSUE THREAD

**Issue by willbuckner** [cockroachdb/cockroach#31368]
**Summary**

`cockroach sql` panics on macOS Mojave with `CCL v2.0.5 @ 2018/08/13 20:55:02 (go1.10.3)` when pressing tab key.

**To Reproduce**

1. Start server with `cockroach start --insecure &`.
2. Run `cockroach sql --insecure`
3. Press the TAB key

**Expected behavior**
Autocompletion of available commands, DB names, etc.

**Additional data / screenshots**

```
*@spacecraft~/go/src/my_project⦕ cockroach sql --insecure
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
#
# Server version: CockroachDB CCL v2.0.5 (x86_64-apple-darwin17.7.0, built 2018/08/13 20:55:02, go1.10.3) (same version as client)
# Cluster ID: 90fb7c2a-3ddc-45cd-b5b8-8a24acb6c6e3
#
# Enter \? for a brief introduction.
#
warning: no current database set. Use SET database = <dbname> to change, CREATE DATABASE to make a new database.
root@:26257/> dyld: lazy symbol binding failed: Symbol not found: _fn_complete
  Referenced from: /usr/local/bin/cockroach
  Expected in: /usr/lib/libedit.3.dylib

dyld: Symbol not found: _fn_complete
  Referenced from: /usr/local/bin/cockroach
  Expected in: /usr/lib/libedit.3.dylib

SIGABRT: abort
PC=0x12d721c2 m=0 sigcode=0

goroutine 0 [idle]:
runtime: unknown pc 0x12d721c2
stack: frame={sp:0x7ffeefbfcf08, fp:0x0} stack=[0x7ffeefb7ee10,0x7ffeefbfe290)
00007ffeefbfce08:  000000005bc4b44d  0000000024e42b08
00007ffeefbfce18:  0000000000000000  0000000000000000
00007ffeefbfce28:  0000000000000000  0000000000000000
00007ffeefbfce38:  0000000000020000  0000000000000000
00007ffeefbfce48:  0000000000000000  0000000000000000
00007ffeefbfce58:  b6272a38c25a0065  0000000009a72980
00007ffeefbfce68:  000000000a099000  0000000009a72980
00007ffeefbfce78:  00007ffeefbfe08c  00007ffeefbfcf60
00007ffeefbfce88:  0000000012d2cc33  0000000012d2fc2b
00007ffeefbfce98:  0000000009a72980  0000000000000000
00007ffeefbfcea8:  0000000012d7290e  0000000000000000
00007ffeefbfceb8:  00007ffeefbfdb70  0000000000000000
00007ffeefbfcec8:  0000000012d2b000  00007ffeefbfdba0
00007ffeefbfced8:  0000000000000000  00007ffeefbfd318
00007ffeefbfcee8:  0000000000000005  0000000009a5a460
00007ffeefbfcef8:  00007ffeefbfd778  00007ffeefbfd7f0
00007ffeefbfcf08: <0000000012d719b9  1b0041363b315b1b
00007ffeefbfcf18:  0000002012d66d37  00007ffeefbfcf70
00007ffeefbfcf28:  0000000000000053  0000000012dade28
00007ffeefbfcf38:  00000000ffffffff  0000000009a72980
00007ffeefbfcf48:  00007ffeefbfcf70  00007ffeefbfcf60
00007ffeefbfcf58:  0000000012d719eb  00007ffeefbfdb90
00007ffeefbfcf68:  0000000012d2faea  6e206c6f626d7953
00007ffeefbfcf78:  646e756f6620746f  6f635f6e665f203a
00007ffeefbfcf88:  200a6574656c706d  6e65726566655220
00007ffeefbfcf98:  6d6f726620646563  6c2f7273752f203a
00007ffeefbfcfa8:  6e69622f6c61636f  616f726b636f632f
00007ffeefbfcfb8:  70784520200a6863  6e69206465746365
00007ffeefbfcfc8:  6c2f7273752f203a  646562696c2f6269
00007ffeefbfcfd8:  6c79642e332e7469  584e6b00000a6269
00007ffeefbfcfe8:  3754584e6b003654  6b00335652506b00
00007ffeefbfcff8:  52506b0034565250  365652506b003556
runtime: unknown pc 0x12d721c2
stack: frame={sp:0x7ffeefbfcf08, fp:0x0} stack=[0x7ffeefb7ee10,0x7ffeefbfe290)
00007ffeefbfce08:  000000005bc4b44d  0000000024e42b08
00007ffeefbfce18:  0000000000000000  0000000000000000
00007ffeefbfce28:  0000000000000000  0000000000000000
00007ffeefbfce38:  0000000000020000  0000000000000000
00007ffeefbfce48:  0000000000000000  0000000000000000
00007ffeefbfce58:  b6272a38c25a0065  0000000009a72980
00007ffeefbfce68:  000000000a099000  0000000009a72980
00007ffeefbfce78:  00007ffeefbfe08c  00007ffeefbfcf60
00007ffeefbfce88:  0000000012d2cc33  0000000012d2fc2b
00007ffeefbfce98:  0000000009a72980  0000000000000000
00007ffeefbfcea8:  0000000012d7290e  0000000000000000
00007ffeefbfceb8:  00007ffeefbfdb70  0000000000000000
00007ffeefbfcec8:  0000000012d2b000  00007ffeefbfdba0
00007ffeefbfced8:  0000000000000000  00007ffeefbfd318
00007ffeefbfcee8:  0000000000000005  0000000009a5a460
00007ffeefbfcef8:  00007ffeefbfd778  00007ffeefbfd7f0
00007ffeefbfcf08: <0000000012d719b9  1b0041363b315b1b
00007ffeefbfcf18:  0000002012d66d37  00007ffeefbfcf70
00007ffeefbfcf28:  0000000000000053  0000000012dade28
00007ffeefbfcf38:  00000000ffffffff  0000000009a72980
00007ffeefbfcf48:  00007ffeefbfcf70  00007ffeefbfcf60
00007ffeefbfcf58:  0000000012d719eb  00007ffeefbfdb90
00007ffeefbfcf68:  0000000012d2faea  6e206c6f626d7953
00007ffeefbfcf78:  646e756f6620746f  6f635f6e665f203a
00007ffeefbfcf88:  200a6574656c706d  6e65726566655220
00007ffeefbfcf98:  6d6f726620646563  6c2f7273752f203a
00007ffeefbfcfa8:  6e69622f6c61636f  616f726b636f632f
00007ffeefbfcfb8:  70784520200a6863  6e69206465746365
00007ffeefbfcfc8:  6c2f7273752f203a  646562696c2f6269
00007ffeefbfcfd8:  6c79642e332e7469  584e6b00000a6269
00007ffeefbfcfe8:  3754584e6b003654  6b00335652506b00
00007ffeefbfcff8:  52506b0034565250  365652506b003556

goroutine 1 [syscall]:
runtime.cgocall(0x543b768, 0xc42080d8a0, 0x5e56868)
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/cgocall.go:128 +0x64 fp=0xc42080d858 sp=0xc42080d820 pc=0x40058a4
github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix._C2func_go_libedit_gets(0xa099000, 0x9a493a0, 0x0, 0x0, 0xc4202c4c08, 0xc4202c4c0c, 0x1, 0x0, 0x0, 0x0)
        _cgo_gotypes.go:175 +0x5c fp=0xc42080d8a0 sp=0xc42080d858 pc=0x53d5f1c
github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix.EditLine.GetLine.func1(0xa099000, 0x9a493a0, 0x0, 0x0, 0xc4202c4c08, 0xc4202c4c0c, 0xc400000001, 0x9a493a02080d9b8, 0x736dc80, 0xc42080d978)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix/editline_unix.go:268 +0xc8 fp=0xc42080d900 sp=0xc42080d8a0 pc=0x53d9c78
github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix.EditLine.GetLine(0x0, 0xe, 0x9a493a0, 0xc42080d9e8, 0x53d8a53)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix/editline_unix.go:268 +0xe1 fp=0xc42080d990 sp=0xc42080d900 pc=0x53d8451
github.com/cockroachdb/cockroach/pkg/cli.(*cliState).doReadLine(0xc42063d040, 0x6, 0x5)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:719 +0x45f fp=0xc42080da20 sp=0xc42080d990 pc=0x540226f
github.com/cockroachdb/cockroach/pkg/cli.runInteractive(0xc4202ae780, 0x0, 0x0)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:1097 +0x68f fp=0xc42080dbf8 sp=0xc42080da20 pc=0x54042cf
github.com/cockroachdb/cockroach/pkg/cli.runTerm(0x73320c0, 0xc4202c6e60, 0x0, 0x1, 0x0, 0x0)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:1179 +0x159 fp=0xc42080dc50 sp=0xc42080dbf8 pc=0x5404869
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0x73320c0, 0xc4202c6e60, 0x0, 0x1, 0x0, 0x0)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:40 +0x5a fp=0xc42080dcc8 sp=0xc42080dc50 pc=0x541ad5a
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0x73320c0, 0xc4202c6e40, 0x1, 0x1, 0x73320c0, 0xc4202c6e40)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:698 +0x46d fp=0xc42080dd70 sp=0xc42080dcc8 pc=0x53a7a5d
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x7334920, 0x8, 0x0, 0xc42080dee8)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:783 +0x2e4 fp=0xc42080dea0 sp=0xc42080dd70 pc=0x53a81d4
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0x7334920, 0x18, 0xc42080df00)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:736 +0x2b fp=0xc42080ded0 sp=0xc42080dea0 pc=0x53a7ecb
github.com/cockroachdb/cockroach/pkg/cli.Run(0xc42003a1c0, 0x2, 0x2, 0xc4200480c8, 0xc420708000)
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:156 +0x6d fp=0xc42080def8 sp=0xc42080ded0 pc=0x53e2a1d
github.com/cockroachdb/cockroach/pkg/cli.Main()
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:51 +0x15d fp=0xc42080df78 sp=0xc42080def8 pc=0x53e25ad
main.main()
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/main.go:27 +0x20 fp=0xc42080df88 sp=0xc42080df78 pc=0x5437150
runtime.main()
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/proc.go:198 +0x212 fp=0xc42080dfe0 sp=0xc42080df88 pc=0x4030832
runtime.goexit()
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc42080dfe8 sp=0xc42080dfe0 pc=0x405d9a1

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/sigqueue.go:139 +0xa7
os/signal.loop()
        /usr/local/Cellar/go/1.10.3/libexec/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
        /usr/local/Cellar/go/1.10.3/libexec/src/os/signal/signal_unix.go:28 +0x41

goroutine 11 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1161 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:592 +0x110

goroutine 12 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:601 +0x105
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
        /private/tmp/cockroach-20180813-63391-1en7bsc/cockroach-v2.0.5/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:593 +0x128

goroutine 50 [select, locked to thread]:
runtime.gopark(0x5f3fac0, 0x0, 0x5e0ca0e, 0x6, 0x18, 0x1)
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420497f50, 0xc4202bc060)
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/signal_unix.go:549 +0x1c6
runtime.goexit()
        /usr/local/Cellar/go/1.10.3/libexec/src/runtime/asm_amd64.s:2361 +0x1

rax    0x2000209
rbx    0x0
rcx    0x7ffeefbfcf08
rdx    0x7ffeefbfd370
rdi    0x6
rsi    0x4
rbp    0x7ffeefbfcf50
rsp    0x7ffeefbfcf08
r8     0x7ffeefbfcf70
r9     0x0
r10    0x53
r11    0x246
r12    0x53
r13    0x7ffeefbfd370
r14    0x4
r15    0x6
rip    0x12d721c2
rflags 0x246
cs     0x7
fs     0x0
gs     0x0
```

**Environment:**
 - CockroachDB version `CCL v2.0.5 @ 2018/08/13 20:55:02 (go1.10.3)`
 - Server OS: macOS Mojave
 - Client app [e.g. `cockroach sql`, JDBC, ...]: cockroach sql

**Additional context**
What was the impact?

`cockroach sql` panic()d

**Comment by knz** [comment]
This has been addressed and will be fixed in a later update to v2.0 and, of course, in v2.1.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by kenliu** [comment]
@knz can this issue be closed now?

**Comment by knz** [comment]
Yes thank you
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by kenliu** [comment]
@knz was this resolved in 19.1?

**Comment by knz** [comment]
No only 19.2. Fix cannot be back ported unfortunately.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
